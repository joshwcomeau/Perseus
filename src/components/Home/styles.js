import { StyleSheet } from '../../helpers/global-aphrodite';

export default StyleSheet.create({
  home: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    color: '#FFF',
  },
});
