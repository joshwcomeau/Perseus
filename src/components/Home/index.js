// eslint-disable-next-line no-unused-vars
import React, { Component, PropTypes } from 'react';
import { css } from '../../helpers/global-aphrodite';

import Logo from '../Logo';
import styles from './styles';

class Home extends Component {
  componentWillMount() {
    this.formElements = {};

    this.submitForm = this.submitForm.bind(this);
  }

  submitForm(ev) {
    ev.preventDefault();

    const user = this.formElements.user.value;
    const repo = this.formElements.repo.value;

    this.context.router.transitionTo(`/examine/${user}/${repo}`);
  }

  render() {
    return (
      <div className={css(styles.home)}>
        <Logo />
        <p>Learn about the people who've starred your repositories.</p>

        <form onSubmit={this.submitForm}>
          <input type="text" ref={elem => this.formElements.user = elem} />
          /
          <input type="text"ref={elem => this.formElements.repo = elem} />
          <button >Go</button>
        </form>
      </div>
    );
  }
};

Home.propTypes = {

};

Home.defaultProps = {

};

Home.contextTypes = {
  router: PropTypes.object,
};

export default Home;
