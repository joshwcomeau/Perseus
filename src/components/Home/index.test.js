/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Home from './index';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
