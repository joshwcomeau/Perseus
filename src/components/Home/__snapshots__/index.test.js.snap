exports[`Home renders correctly 1`] = `
<div
  className="home_117m6hr">
  <h2
    className="logo_1d4z0p1">
    Perseus
  </h2>
  <p>
    Learn about the people who\'ve starred your repositories.
  </p>
  <form
    onSubmit={[Function]}>
    <input
      type="text" />
    /
    <input
      type="text" />
    <button>
      Go
    </button>
  </form>
</div>
`;
